### ISSUE TITLE
@rjsf/core/dist/validate module lost when minor number changed, v4.1.0 -> v.4.2.0

### ISSUE THREAD

**Issue by postk** [rjsf-team/react-jsonschema-form#2966]
### Prerequisites

- [X] I have searched the existing issues
- [X] I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/)
- [ ] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### What theme are you using?

core

### Version

4.2.0 and higher

### Current Behavior

Since v4.2.0 `lib/validate.js` file does not included in core package build

### Expected Behavior

`lib/validate.js` file should be included in core package build in v4.2.0, as it is described in `index.d.ts`

### Steps To Reproduce

Update core package dependency version from 4.1.0
```
{
  "dependencies": {
    "@rjsf/core": "^4.1.0"
  }
}
```
to version 4.2.0
```
{
  "dependencies": {
    "@rjsf/core": "^4.2.0"
  }
}
```

### Environment

```markdown
- OS: any
- Node: any
- npm: any
```

### Anything else?

In fact, part of functionality provided by package has been removed _(in version 4.2.0, previous version is 4.1.0)_, this is breaking change. According to semantic versioning, breaking changes should be made when major number changes, but it is done when minor number is changed

Here is example of `dist` folder of v4.1.0 ![v4.1.0](https://user-images.githubusercontent.com/70149836/181590937-c7fc57cb-1ff1-4964-958d-5927c64cc78a.png)

And here is example of `dist` folder of v4.2.0
![v4.2.0](https://user-images.githubusercontent.com/70149836/181591121-51ff9ec3-cd49-432d-9882-32b243aa5c87.png)

At the same time module `@rjsf/core/lib/validate` described in `index.d.ts` file for both versions, what is really wrong

**Comment by heath-freenome** [comment]
Have you checked v4.2.2?

**Comment by postk** [comment]
just checked, issue is there

__steps to reproduce:__
- download source code [https://github.com/rjsf-team/react-jsonschema-form/releases/tag/v4.2.2](https://github.com/rjsf-team/react-jsonschema-form/releases/tag/v4.2.2)
- install dependencies
- build packages

__result:__ lib folder lost in core package build

__note:__
actually `validate.js` file stored in `src` folder, by this path: `/packages/core/src/validate.js`
but `src` folder does not included in package build..

![ex__mark](https://user-images.githubusercontent.com/70149836/182321805-913fba36-3180-4dc2-b0bb-982e7ac79caa.png)

**Comment by postk** [comment]
Seems issue happened [here](https://github.com/rjsf-team/react-jsonschema-form/commit/724fb0bc3f1ae5d1d9b761b2bff7cf5c64fd7459#diff-0b810c38f3c138a3d5e44854edefd5eb966617ca84e62f06511f60acc40546c7R6).
Suggest to replace default export with named one in `validate.js` and re-export everything from `index.js`, like it is done for utils.
Here is [patch](https://gist.github.com/postk/4b61e92b145b9d583695a68570d77476), tests are fine:
```
Test Suites: 23 passed, 23 total
Tests:       1390 passed, 1390 total
Snapshots:   0 total
Time:        13.233s
```

**Comment by rodrigoaguilera** [comment]
Is this related to the disappearance of `core/dist/cjs/utils.js` from 4.2.0 to 4.2.2?

I have this in my code 
```
import { isMultiSelect, getUiOptions, isFilesArray } from '@rjsf/core/dist/cjs/utils';
```
and I can't update the package.

**Comment by postk** [comment]
@rodrigoaguilera, sorry, my bad, mixed up versions in title. It is related to disappearance of `@rjsf/core/lib/validate`. It is included in package build v.4.2.0 and lower. Disappeared in 4.2.1 and higher.

```
import validateFormData from '@rjsf/core/lib/validate';
```

**Comment by nickgros** [comment]
@postk can you confirm if this has been fixed in v4.2.3?

**Comment by postk** [comment]
@nickgros, yes, thank you, v4.2.3 looks good
